### ISSUE TITLE
Accessibility Improvements ‣ Shop ‣ Action Buttons

### ISSUE THREAD

**Issue by barryhughes** [woothemes/woocommerce#37601]
As originally noted [over here](https://github.com/woocommerce/woocommerce/issues/24185#issuecomment-779227823), the **Select Options** (variable products) and **Add to Cart** buttons do not have meaningful text.

<div align="center"><img width="480" alt="shop-selectoptions-addtocart" src="https://user-images.githubusercontent.com/3594411/230445655-5c2f1af0-f969-41b0-8446-7c3c2a673405.png"></div>

It is proposed we add an `aria-describedby` attribute to these buttons, referencing the corresponding product.

**Comment by faisal-alvi** [comment]
Hi @opr 👋🏻 I would like to work on this.

However, I can see the `aria-label` already has a proper description of the object.

For example,

* `aria-label="Select options for “Hoodie"`
* `aria-label="Add “Beanie” to your cart"`

Based on our previous discussion in #37599, it doesn't appear to be necessary to add the `aria-describedby` attribute to the buttons. What do you think?

**Comment by opr** [comment]
Hi @faisal-alvi - yes I agree with you for the regular "add to cart" button, however I wonder if adding a descriptions for the "select options" buttons would be a good idea? This description could mention what will happen after clicking.

Just an example: "This product has multiple variants. The options may be chosen on the product page"

**Comment by faisal-alvi** [comment]
@opr Yes, we can add a description to the `Select options` button.

We have 2 options.

1. We can add the description in the `aria-label` attribute.

`aria-label="Select options for \"Hoodie\". It has multiple variants. The options may be chosen on the product page"`

2. Or we add `aria-describedby` attribute and add the description there.

`aria-describedby="It has multiple variants. The options may be chosen on the product page"`

Both attributes are used together, the screen reader will first read the label specified in `aria-label`, followed by the additional description provided in `aria-describedby`.

**Comment by opr** [comment]
Hey @faisal-alvi thank you for describing the two options! I think option 2 would be the best approach here, however "It has..." does not read super well, please could we try with "This product has..."?

Also, we would only need one element to describe the "Select Options" button - saving a little bit of data compared to adding it to the label.
